src/EditorFeatures/CSharpTest/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -11185,5 +11185,36 @@ private static bool NewMethod()
 }";
             await TestExtractMethodAsync(code, expected);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task ExtractMethodInvolvingFunctionPointer()
+        {
+            var code = @"
+class C
+{
+    void M(delegate*<delegate*<ref string, ref readonly int>> ptr1)
+    {
+        string s = null;
+        _ = [|ptr1()|](ref s);
+    }
+}";
+
+            var expected = @"
+class C
+{
+    void M(delegate*<delegate*<ref string, ref readonly int>> ptr1)
+    {
+        string s = null;
+        _ = NewMethod(ptr1)(ref s);
+    }
+
+    private static delegate*<ref string, ref readonly int> NewMethod(delegate*<delegate*<ref string, ref readonly int>> ptr1)
```

Comment:
lordy those look awful :D

Reply:
Honestly, I don't think it's that bad. It's very similar to `Func`. And certainly a heck of a lot better than trying to put actual method syntax in there (which doesn't feel very C# to me at all).